app/registries/foreman/access_control.rb

```diff
@@ -23,6 +23,16 @@ def map
         yield mapper
         @permissions ||= []
         @permissions += mapper.mapped_permissions
+        mapper.actions_to_add.each do |perm_name, actions_with_controller|
+          p = permission(perm_name)
+          allowed_actions = []
+          actions_with_controller.map do |controller, actions|
+            actions.each do |action|
+              allowed_actions << "#{controller}/#{action}"
+            end
+          end
+          p.actions.concat(allowed_actions)
+        end
```

Comment:
Seeing this, wouldn't it be easier to just to initialize the Permission in every case and on the collision just have `p.actions.concat(collision_perm.actions)`?

We could even ignore collisions in the mapper and resolve them here.
```ruby
@permissions ||= []
add_permissions(mapper.mapped_permissions)
add_permissions(mapper.added_permission_actions, true)

def add_permissions(permissions, adding = false)
  permissions.each do |perm|
    if (p = permission(perm.name))
      # Deprecation if adding
      p.actions.concat(perm.actions)
    else
      # error perm missing if !adding
      @permissions << perm
    end
  end
end
```